### ISSUE TITLE
[Bug]: Socket connection to 'wss://domain.com/ng-cli-ws' failed: in chrome's developer console when running angular in code-server

### ISSUE THREAD

**Issue by Vaibhav1919** [coder/code-server#6205]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS/Web Information

- Web Browser: Chrome
- Local OS: Ubuntu
- Remote OS: Ubuntu
- Remote Architecture: amd64
- `code-server --version`: 4.12.0

### Steps to Reproduce

1. Deploy code-server as pod and create service in EKS cluster
2. Use Emissary-ingress as proxy and create mapping to code-server userpod (eg /tempmap/ --> userpod service)
3. Access the userpod at the URL www.domain.tld/tempmap/
4. Install Nvm , npm and angular
5. switch to the created directory and do ng serve and access the browser
6. Now we will get white screen error
7. Follow the code-server proxy guidelines
8. try  a) use <base href="./."> in src/index.html
 b) add --serve-path /absproxy/4200 to ng serve in your package.json
9. Socket connection to 'wss://domain.tld/ng-cli-ws' failed: not found error in chrome's developer console

### Expected

Not getting Socket connection to 'wss://domain.tld/ng-cli-ws' failed:  error in chrome developer consoleWebSocket

### Actual

Getting webSocket connection to 'wss://domain.tld/ng-cli-ws' failed error 

### Does this issue happen in VS Code or GitHub Codespaces?

- [X] I cannot reproduce this in VS Code.
- [X] I cannot reproduce this in GitHub Codespaces.

### Are you accessing code-server over HTTPS?

- [X] I am using HTTPS.

### Notes

Code-server is deployed on AWS EKS cluster, and I am using emissary-ingress as proxy to access code-server. SSL is attached to AWS application load balancer used to access code-server. We are getting this error not always but getting them randomly.

**Comment by code-asher** [comment]
Looks like Angular is dropping the base path and going against the root but unfortunately I am not familiar enough with Angular to say why but you might want to look at `--base-href` or `--deploy-url` as well.

**Comment by code-asher** [comment]
Hmm so I think the problem might be a reverse proxy in front of code-server's proxy.  If I serve code-server at `domain.tld/some-path` then try to access `domain.tld/some-path/absproxy/4200/` then the first reverse proxy rewrites to `/absproxy/4200`/ which is all code-server sees and it passes that along to Angular but this is not right because it drops `some-path`.

So the solution would be to prevent the first proxy from rewriting but if you do that then code-server itself will not work because code-server does expect rewriting.  You would get a 404 since `/some-path` does not exist.

Ultimately I am not sure this can be resolved.  code-server expects rewriting and Angular expects no rewriting from the proxy so they are fundamentally at odds.  You would probably need to either host code-server at the domain root or a sub-domain.

**Comment by code-asher** [comment]
Actually still not working...it times out now.  Still not sure this can be resolved without some fix upstream.

**Comment by Vaibhav1919** [comment]
Hi @code-asher  after adding this <base href="./."> in src/index.html angular app is working but i am getting socket connection failed error.

**Comment by code-asher** [comment]
Yeah mine times out.  I have no idea what causes it unfortunately.  It seems to be the right path.
